tests/test_core.py

```diff
@@ -36,22 +36,24 @@ def test_core_tables(mock_db_core, table):
 
     # The schema check is to ensure we have a consistent order for the data in
     # these files, mostly for making git history simpler in case of minor changes
-    table_rows = conftest.get_sorted_table_data(cursor, table)
-
+    table_rows, cols = conftest.get_sorted_table_data(cursor, table)
     # For regenerating data if needed
     # with open(f"./tests/test_data/core/{table}.txt", "wt", encoding="UTF-8") as f:
     #     for row in table_rows:
     #         f.write(str(f"{row}\n"))
-
     with open(f"./tests/test_data/core/{table}.txt", encoding="UTF-8") as f:
         ref_table = []
         for row in f.readlines():
             ref_table.append(eval(row))  # pylint: disable=eval-used
-    for row in ref_table:
-        assert row in table_rows
-    for row in table_rows:
-        assert row in ref_table
-    assert len(table_rows) == len(ref_table)
+    try:
+        for row in ref_table:
+            assert row in table_rows
+        for row in table_rows:
+            assert row in ref_table
+        assert len(table_rows) == len(ref_table)
+    except Exception as e:
+        conftest.debug_diff_tables(cols, table_rows, ref_table, pos=0)
+        raise e
```

Comment:
i'm leaving this in permanently for error states because it is incredibly useful for troubleshooting data problems.

Reply:
Fail Fast, love it!